src/scripts/main.js

```diff
@@ -1,3 +1,69 @@
 'use strict';
 
-// write code here
+const appendRowButton = document.querySelector('.append-row');
+const removeRowButton = document.querySelector('.remove-row');
+const appendColumnButton = document.querySelector('.append-column');
+const removeColumnButton = document.querySelector('.remove-column');
+const table = document.querySelector('.field');
+
+// let totalRows = table.rows.length;
+// let columns = table.rows[0].cells.length;
+
+function updateButtonStates() {
+  const rowCount = table.rows.length;
+  const columnCount = table.rows[0].cells.length;
+
+  if (rowCount === 10) {
+    appendRowButton.disabled = true;
+  } else if (rowCount === 2) {
+    removeRowButton.disabled = true;
+  } else {
+    appendRowButton.disabled = false;
+    removeRowButton.disabled = false;
```

Comment:
```suggestion
    appendRowButton.disabled = rowCount === 10;
    removeRowButton.disabled = rowCount === 2;
    
    appendColumnButton.disabled = columnCount === 10;
    removeColumnButton.disabled = columnCount === 2;
    
```
let's simplify your function and get rid of if else structure